token/transfer-hook-interface/src/offchain.rs

```diff
@@ -0,0 +1,64 @@
+//! Offchain helper for fetching required accounts to build instructions
+
+use {
+    crate::{get_extra_account_metas_address, instruction::ExecuteInstruction},
+    solana_sdk::{
+        account::Account, instruction::AccountMeta, program_error::ProgramError, pubkey::Pubkey,
+    },
+    spl_tlv_account_resolution::state::ExtraAccountMetas,
+    std::future::Future,
+};
+
+type AccountResult = Result<Option<Account>, AccountFetchError>;
+type AccountFetchError = Box<dyn std::error::Error + Send + Sync>;
+
+/// Offchain helper to get all additional required account metas for a mint
+///
+/// To be client-agnostic, this simply takes a function that will return a
+/// `Future<Account>` for the given address. Can be called in the following way:
+///
+/// ```rust,ignore
+/// use solana_client::nonblocking::rpc_client::RpcClient;
+/// use solana_sdk::pubkey::Pubkey;
+///
+/// let program_id = Pubkey::new_unique();
+/// let mint = Pubkey::new_unique();
+/// let client = RpcClient::new_mock("succeeds".to_string());
+///
+/// let extra_account_metas = get_extra_account_metas(
+///     |address| self.client.get_account(&address),
```

Comment:
The point is that off-chain clients can infer their own PDAs, right?

Reply:
In a generic library, this makes sense. But in this library, I think this is too general, and a bit confusing. 
 
I think here it should only take `mint`.

IMO it can be spun out onto a library ontop of tlv in new lib like `tlv-account-resolution-keyed` ¯\_(ツ)_/¯ just spitballing here.

Reply:
The point of this is to decouple it from any particular client, so it works with rpc-client and banks-client, which is how the token-client works. So at its core this only takes the mint and program id, which are the only required bits to fetch the info. In the implementation, it does the PDA derivation itself. It felt simpler to keep it in this library with a feature flag rather than spinning out a separate crate just for one file.

Reply:
Checks out, thanks!